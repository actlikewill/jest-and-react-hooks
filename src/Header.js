import React from 'react'

export default function Header ({ text }) {
    return (
        <header>
            <h1 data-testid="header-tag" className="headerStyle">{text}</h1>
        </header>
    )
}