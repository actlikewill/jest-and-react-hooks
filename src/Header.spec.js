import React from 'react'
import { render } from '@testing-library/react';
import Header from './Header'

it('renders the header', () => {
    const {asFragment} = render(<Header text='Hello' />)
    expect(asFragment()).toMatchSnapshot()
})

it('renders the expected text', () => {
    const {getByTestId, getByText} = render(<Header text='Hello' />)
    expect(getByTestId('header-tag')).toHaveTextContent('Hello')
    expect(getByText('Hello')).toHaveClass('headerStyle')

})