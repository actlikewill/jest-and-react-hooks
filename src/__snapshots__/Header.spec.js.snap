// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the header 1`] = `
<DocumentFragment>
  <header>
    <h1
      class="headerStyle"
      data-testid="header-tag"
    >
      Hello
    </h1>
  </header>
</DocumentFragment>
`;
